Sources/NIO/ConvenienceOptionSupport.swift

```diff
@@ -0,0 +1,186 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// MARK:  Universal Client Bootstrap
+extension NIOClientTCPBootstrapProtocol {
+    /// Apply any understood shorthand options to the bootstrap, removing them from the set of options if they are consumed.
+    /// - parameters:
+    ///     - options:  The options to try applying - the options applied should be consumed from here.
+    /// - returns: The updated bootstrap with and options applied.
+    public func _applyChannelConvenienceOptions(_ options: inout ChannelOptions.NIOTCPShorthandOptions) -> Self {
+        // Default is to consume no options and not update self.
+        return self
+    }
+}
+
+extension NIOClientTCPBootstrap {
+    /// Specifies some `ChannelOption`s to be applied to the channel.
+    /// - See: channelOption
+    /// - Parameter options: Set of shorthand options to apply.
+    /// - Returns: The updated bootstrap (`self` being mutated)
+    public func channelConvenienceOptions(_ options: ChannelOptions.NIOTCPShorthandOptions) -> NIOClientTCPBootstrap {
+        var optionsRemaining = options
+        // First give the underlying a chance to consume options.
+        let withUnderlyingOverrides =
+            NIOClientTCPBootstrap(self, withUpdated:
+                                    underlyingBootstrap._applyChannelConvenienceOptions(&optionsRemaining))
+        // Default apply any remaining options.
+        return optionsRemaining.applyFallbackMapping(withUnderlyingOverrides)
+    }
+}
+
+// MARK: Utility
+extension ChannelOptions.Types {
+    /// Has an option been set?
+    /// Option has a value of generic type T.
+    public enum NIOOptionValue<T> {
```

Comment:
- doesn't need the `NIO` prefixed because now nested
- should contain "convenience". So maybe `ConvenienceOptionValue`?
- `T` isn't a great generic parameter name, maybe `Value` or `ValueType` or so?